source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -40,9 +41,34 @@ namespace
     static const RtToken OUTPUT_ELEMENT_PREFIX("OUT_");
     static const RtToken MULTIOUTPUT("multioutput");
 
-    PvtPrim* findPrimOrThrow(const RtToken& name, PvtPrim* parent)
+    class PvtRenamingMapper {
+        typedef RtTokenMap<RtToken> TokenToToken;
+        typedef std::map<PvtPrim*, TokenToToken> PerPrimMap;
+
+        PerPrimMap _map;
+    public:
+        void addMapping(PvtPrim* parent, const RtToken& originalName, const RtToken& finalName) {
+            if (originalName != finalName) {
+                _map[parent][originalName] = finalName;
+            }
+        }
+
+        const RtToken& getFinalName(PvtPrim* parent, const RtToken& originalName) const {
+            PerPrimMap::const_iterator primTarget = _map.find(parent);
+            if (primTarget != _map.cend()) {
+                TokenToToken const& nameMap = primTarget->second;
+                TokenToToken::const_iterator nameTarget = nameMap.find(originalName);
+                if (nameTarget != nameMap.cend()) {
+                    return nameTarget->second;
+                }
+            }
+            return originalName;
+        }
+    };
+
+    PvtPrim* findPrimOrThrow(const RtToken& name, PvtPrim* parent, PvtRenamingMapper const& mapper)
```

Comment:
Nitpick: our convention is to put const to the left of the type: `const PvtRenamingMapper& mapper`